cypress/integration/products_page.spec.js

```diff
@@ -1,74 +1,56 @@
-describe('Products Page', () => {
+import {addProduct, visitPage, checkcomponents, checkBasketTotal, checkBasketLength, cleanTheBasket} from '../pageObjects/products_page_object';
+import * as Element from '../pageObjects/constElements';
+import {toogleQuickView, quickViewIsVisible, quickViewIsNotVisible} from '../pageObjects/quick_view_pageObjects';
 
-    it('should visit the product page', () => {
-      cy.visit('http://localhost:4200/');
+
+describe('Products Page', () => {
   
-      cy.get('app-products')
-        .should('exist');
-      cy.get('app-quickview')
-        .should('exist');
-      cy.get('app-basket')
-        .should('exist');
+    it('should visit the product page', () => {
+      visitPage();
+      
+      checkcomponents();
     });
   
     it('should add product to the basket', () => {
-      cy.get('[id="addToBasketButton-0"]')
-        .click();
-    
-      cy.get('[id="basketTotal"]')
-        .should('not.contain', '0');
-      cy.get('[id="basketLength"]')
-        .should('contain', '1');
+      addProduct(Element.productOne);
+
+      checkBasketTotal(Element.notContain, Element.valueZero);
```

Comment:
I would question if there really need to be variable for 0 :). I think it is more readable if it is just plain 0.
Also the first argument is an assertion, so personally I find it confusing that it is coming from Element